### ISSUE TITLE
Optimize scheduler logic and remove nodelock

### ISSUE THREAD

**Issue by lengrongfu** [Project-HAMi/HAMi#450]
Currently, there are some problems with using nodelock that cannot be solved.
- https://github.com/Project-HAMi/HAMi/issues/404
- https://github.com/Project-HAMi/HAMi/issues/382
- https://github.com/Project-HAMi/HAMi/issues/449

Current hami scheduler use nodelock to implement serial bind pod, and release the nodelock only after device-plugin allocate device succeeds, but

**Comment by archlitchi** [comment]
yes, it seems ascend-for-volcano use a FIFO strategy with 'bind-time timestamp' instead of using nodelock, and works properly, we can try that as well
